### ISSUE TITLE
Every `*-single-line` and `*-multi-line` option reads a construct by its line feeds and carriage returns alone, a form feed counting for neither

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#244]
Found while fixing [#209](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/209), which moved the two readings of `block-closing-brace-newline-before` to the set of line breaks the rest of the plugin reads. That rule has a third reading, and it is shared with most of the plugin: whether the thing it is looking at is multi-line at all.

### What minimal example or steps are needed to reproduce the bug?

A construct whose only line break is a form feed:

```css
a { b: fn(\fc\f); }
a {color: pink;\ftop: 0;}
a\f,b {}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/function-parentheses-space-inside": "always-single-line",
    "@stylistic/block-opening-brace-newline-after": "always-multi-line",
    "@stylistic/selector-list-comma-space-after": "always-single-line"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json --fix
```

### What did you expect to happen?

The `\n` twin of each file decides the other way: `a { b: fn(\nc\n); }` is a multi-line function and the single-line option passes it over, `a {color: pink;\ntop: 0;}` is a multi-line block and draws `Expected newline after "{" of a multi-line block`, and `a\n,b {}` is not a single-line list and draws nothing. A form feed ends a line to every syntax this plugin reads through, and [#173](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/173), [#196](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/196), [#204](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/204) and [#209](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/209) each moved a reading to `[\n\r\f]` for that reason, so the same three answers.

### What actually happened?

Every one of the three is read the other way round, and the harm runs in both directions.

A `*-multi-line` option passes the file over in silence: `a {color: pink;\ftop: 0;}` draws nothing from `block-opening-brace-newline-after`, `a {\fcolor: pink;\f}` nothing from `block-closing-brace-empty-line-before`, and `a { b: fn(c,\fd); }` nothing from `function-parentheses-newline-inside`.

A `*-single-line` option reports a construct that holds a break, and its fix then writes over that break:

```
a { b: fn(\fc\f); }   with function-parentheses-space-inside: "always-single-line"
 --fix  a { b: fn( c ); }
```

Both form feeds are gone, and two lines of the file have become one. `selector-list-comma-space-after` and `declaration-block-semicolon-space-after` report the same way, though their fixes write beside the break rather than over it.

### What is the cause?

[`isSingleLineString`](https://github.com/stylelint-stylistic/stylelint-stylistic/blob/main/lib/utils/isSingleLineString/index.js) tests its input with `CR_OR_LF`, `/[\r\n]/u`, and that is the whole of it. It is what [`whitespaceChecker`](https://github.com/stylelint-stylistic/stylelint-stylistic/blob/main/lib/utils/whitespaceChecker/index.js) asks for every `*-single-line` and `*-multi-line` option, so thirty-one rules ask it through the checker, and `block-closing-brace-empty-line-before`, `block-closing-brace-newline-before`, `function-parentheses-newline-inside` and `function-parentheses-space-inside` ask it directly.

[#173](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/173) moved `selector-pseudo-class-parentheses-space-inside` off that reading to `LINE_BREAK`, `/[\n\r\f]/u`, for exactly this reason, and left the shared util where it was. Moving the util answers for every rule at once.

Two other readings of `CR_OR_LF` are not this question and do not move: `linebreaks` is about which of the two characters a file ends its lines with, and `OPENS_WITH_CR_OR_LF` in `no-empty-first-line` is about the blank first line an editor writes.
